src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -74,6 +74,20 @@ private async Task AnalyzeDocumentForKindAsync(Document document, AnalysisKind k
         }
 
         public async Task AnalyzeProjectAsync(Project project, bool semanticsChanged, InvocationReasons reasons, CancellationToken cancellationToken)
+        {
+            // Perf optimization. check whether we want to analyze this project or not.
+            if (!Executor.FullAnalysisEnabled(project, forceAnalyzerRun: false))
+            {
+                return;
+            }
```

Comment:
This is checked later in the existing AnalyzeProjectAsync code path. However, that check is only for computing new data. We need to add this new check here otherwise the invoked `AnalyzeProjectAsync` invocations from background analysis immediately resets the diagnostics computed from prior Run Code Analysis command execution.